src/main/java/dev/joopie/jambot/service/SpotifyAPIConverterService.java

```diff
@@ -0,0 +1,97 @@
+package dev.joopie.jambot.service;
+
+import dev.joopie.jambot.models.Artist;
+import dev.joopie.jambot.models.Track;
+import dev.joopie.jambot.models.album.Album;
+import dev.joopie.jambot.models.album.AlbumGroup;
+import dev.joopie.jambot.models.album.AlbumType;
+import dev.joopie.jambot.repository.album.AlbumRepository;
+import dev.joopie.jambot.repository.artist.ArtistRepository;
+import dev.joopie.jambot.repository.track.TrackRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import se.michaelthelin.spotify.model_objects.specification.AlbumSimplified;
+import se.michaelthelin.spotify.model_objects.specification.ArtistSimplified;
+
+import javax.xml.bind.ValidationException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class SpotifyAPIConverterService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpotifyAPIConverterService.class);
+    @Autowired
+    private ArtistRepository artistRepository;
+
+    @Autowired
+    private AlbumRepository albumRepository;
+
+    @Autowired
+    private TrackRepository trackRepository;
+
+    public Track saveAPIResult(se.michaelthelin.spotify.model_objects.specification.Track trackresult) {
+        Track track = trackRepository.find().byExternalId(trackresult.getId());
+        if (track == null) {
```

Comment:
Voorkom nesten van code.

Break/return early.

In dit geval, als je al een track hebt, return die, anders ga door met code om alsnog op te halen.